app/services/reset_order_service.rb

```diff
@@ -0,0 +1,23 @@
+class ResetOrderService
+  def initialize(controller, order)
+    @controller = controller
+    @distributor = order.distributor
+    @token = order.token
+  end
+
+  def call
+    controller.expire_current_order
```

Comment:
Is this implemented in all the controllers tha call this service? `Spree::PaypalController` for instance